server/gendb/tests.go.tmpl

```diff
@@ -327,6 +330,193 @@ func Get{{ $modelName }}sBy{{ $indexName }}(d db.Interface, t *testing.T) func(t
 }
 {{- end }}
 
+{{- if $xdbConfig.AllowPrimaryIndexScan }}
+type scan{{ $modelName }}sInput struct {
+	ctx   context.Context
+	input db.Scan{{ $modelName }}sInput
+}
+type scan{{ $modelName }}sOutput struct {
+	{{ varname $modelName }}s []models.{{ $modelName }}
+	err    error
+}
+type scan{{ $modelName }}sTest struct {
+	testName string
+	d        db.Interface
+	input    scan{{ $modelName }}sInput
+	output   scan{{ $modelName }}sOutput
+}
+
+func (g scan{{ $modelName }}sTest) run(t *testing.T) {
+	{{ varname $modelName }}s := []models.{{ $modelName }}{}
+	err := g.d.Scan{{ $modelName }}s(g.input.ctx, g.input.input, func(m *models.{{ $modelName }}, last bool) bool {
+		{{ varname $modelName }}s = append({{ varname $modelName }}s, *m)
+		return true
+	})
+	var errStr string
+	if err != nil {
+		errStr = err.Error()
+	}
+	require.Equal(t, g.output.err, err, errStr)
+	require.Equal(t, g.output.{{ varname $modelName }}s, {{ varname $modelName }}s)
+}
+
+func Scan{{ $modelName }}s(d db.Interface, t *testing.T) func(t *testing.T) {
+	return func(t *testing.T) {
+		ctx := context.Background()
+		require.Nil(t, d.Save{{ $modelName }}(ctx, models.{{ $modelName }}{
+			{{- range $attributeName := $pkModelAttributeNames }}
+			{{ pascalize $attributeName }}:    {{ exampleValueForAttribute $xdbConfig $attributeName 1 }},
```

Comment:
This space after the colon looks weird to me. I see it it other places too - intentional?

Reply:
yeah the formatting here is a mix of spaces and tabs, since a lot of the text is copy-pasted Go code